sysdeps/riscv/dl-trampoline.S

```diff
@@ -56,6 +71,18 @@ ENTRY (_dl_runtime_resolve)
   REG_L a5, 6*SZREG(sp)
   REG_L a6, 7*SZREG(sp)
   REG_L a7, 8*SZREG(sp)
+
+#ifndef __riscv_float_abi_soft
+  FREG_L fa0, (10*SZREG + 0*SZFREG)(sp)
+  FREG_L fa1, (10*SZREG + 1*SZFREG)(sp)
+  FREG_L fa2, (10*SZREG + 2*SZFREG)(sp)
+  FREG_L fa3, (10*SZREG + 3*SZFREG)(sp)
+  FREG_L fa4, (10*SZREG + 4*SZFREG)(sp)
+  FREG_L fa5, (10*SZREG + 5*SZFREG)(sp)
+  FREG_L fa6, (10*SZREG + 6*SZFREG)(sp)
+  FREG_L fa7, (10*SZREG + 7*SZFREG)(sp)
+#endif
+
   addi sp, sp, FRAME_SIZE
 
   # Invoke the callee.
```

Comment:
There was a discussion about this somewhere, and IIRC we don't need it.  We just need to make sure that the dynamic resolver doesn't using floating-point.  In an ideal world we would just compile it with soft float, but there appears to be no easy way to enforce that in glibc.

I think this is safe, so I'm OK going with it.